package.json

```diff
@@ -58,6 +58,7 @@
     "build": "npm run build:ts",
     "release": "release-script",
     "lint": "eslint \"packages/*/*.js\" \"packages/*/lib/**/*.js\"",
+    "prettier": "node node_modules/prettier/bin-prettier.js --config ./.prettierrc.json --write --ignore-path ./.prettierignore \"./packages/**/*.js\"",
```

Comment:
I would make the task call eslint --fix, because eslint will fix prettier errors too and will also show you all unfixable lines, so you have both checks and can call it pre commit

Reply:
I suggest just including Prettier in ESLint and configuring the editors to auto-format on save. No need to run an additional task and this way, linting will fail if a file is mis-formatted.

Not sure how that works in WebStorm, but here's the same for VSCode:
https://github.com/zwave-js/node-zwave-js/blob/cfbbd4c86c86b73c075472abb917efc73dbe6ac2/.vscode/settings.json#L6-L13

Don't forget to include Prettier as a recommended extension:
https://github.com/zwave-js/node-zwave-js/blob/cfbbd4c86c86b73c075472abb917efc73dbe6ac2/.vscode/extensions.json#L4

Reply:
That is exactly how I do it locally and told @Apollon77. And this is also the reason why I recommended to use eslint --fix instead, because that is what the IDE does on saving. However, we have currently no IDE configuration files in the repo and I am not sure if we should add such or keep it to the dev. GH actions will fail in all cases if prettier/linter not applied

Reply:
> However, we have currently no IDE configuration files in the repo and I am not sure if we should add such or keep it to the dev

IMO it makes sense for the common two IDEs. IMO that's way less friction than having to explain contributors how to fix formatting issues.

Reply:
For me it would be fine to check in `.idea` and `.vscode` folders if everyone agrees

Reply:
> For me it would be fine to check in `.idea` and `.vscode` folders if everyone agrees

Till now I found this as a bad practice, as everyone has different preferences.

Reply:
> everyone has different preferences

You'd only configure the repository-specific settings though, like auto-formatting. Not sure about WebStorm, but VSCode distinguishes between user settings and repo settings.

Reply:
.idea folder has a predefined gitignore in it so it should be about equal to vscode

Reply:
> I would make the task call eslint --fix, because eslint will fix prettier errors too and will also show you all unfixable lines, so you have both checks and can call it pre commit

Do it